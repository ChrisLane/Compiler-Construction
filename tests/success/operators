// This tests all operators. This should parse.
main() {
    x = a + b;
    x = a - b;
    x = a * b;
    x = a / b;
    x = a <= b;
    x = a >= b;
    x = a == b;
    x = a != b;
    x = a && b;
    x = a || b;
    x = !(a || b);
}
