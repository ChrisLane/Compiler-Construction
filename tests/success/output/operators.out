main (  ) { 
    Asg ( "x" := Operator ( Plus, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Minus, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Times, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Divide, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Leq, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Geq, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Equal, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Noteq, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( And, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Or, "a" , "b" ) ) ; 
    Asg ( "x" := Operator ( Not, Operator ( Or, "a" , "b" ) ) ) 
}
